subprojects/core/src/test/groovy/org/gradle/internal/service/scopes/ToPlannedNodeConverterRegistryTest.groovy

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.service.scopes
+
+import org.gradle.execution.plan.Node
+import org.gradle.execution.plan.TaskDependencyResolver
+import org.gradle.execution.plan.TaskNode
+import org.gradle.execution.plan.ToPlannedNodeConverter
+import org.gradle.execution.plan.ToPlannedTaskConverter
+import spock.lang.Specification
+
+class ToPlannedNodeConverterRegistryTest extends Specification {
+
+    def "can be empty"() {
+        def registry = new ToPlannedNodeConverterRegistry([])
+        expect:
+        registry.getConverter(Stub(Node)) == null
+    }
+
+    def "finds converter matching type"() {
+        def testConverter = Stub(ToPlannedNodeConverter) {
+            getSupportedNodeType() >> TestNode
+        }
+        def registry = new ToPlannedNodeConverterRegistry([testConverter])
+
+        when:
+        def foundConverter = registry.getConverter(new TestNode())
+        then:
+        foundConverter.is(testConverter)
```

Comment:
🤔 Should we query twice here to check that the caching works?

Reply:
Done